util/topgen/validate.py

```diff
@@ -851,7 +855,7 @@ def check_modules(top, prefix):
                     if mem_type == "flash":
                         check_keys(value['config'], eflash_required, eflash_optional,
                                    eflash_added, "Eflash")
-                        flash = Flash(value['config'])
+                        flash = Flash(value['config'], m['base_addrs'][intf])
```

Comment:
The check in line 866 should be performed before this line, probably in line 851.

Reply:
I moved the check above and changed it to raise an exception before reaching the usage of the entry.